elastic/apm-server
# Add `data_streams.wait_for_integration`

## Motivation/summary

Add `apm-server.data_streams.wait_for_integration` config. When both this and `apm-server.data_streams.enabled` are true, the server will wait at startup until the APM integration package is installed. To do this, APM Server requires `apm-server.kibana` to be configured, and will query the Fleet API to query the integration package status.

## Checklist


For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

1. Create a fresh cluster with no integration isntall
2. Run apm-server with `apm-server.data_streams.enable=true` (`wait_for_integration` defaults to true)
3. Observe that apm-server logs that it is waiting for the integration, and it does not start listening on 8200
4. Install the apm integration package in Kibana
5. Observe that apm-server continues, and starts listening on 8200
6. Send some events
7. Check that they are indexed into data streams

## Related issues

https://github.com/elastic/apm-server/issues/5914
